services/rule_engine/src/main/java/agents/Ava.java

```diff
@@ -1,25 +1,149 @@
 package agents;
 
-import java.util.ArrayList;
-
 import lombok.Data;
-import lombok.NoArgsConstructor;
 import lombok.ToString;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashMap;
 
+import com.mindsmiths.mitems.Mitems;
+import com.mindsmiths.mitems.Option;
 import com.mindsmiths.ruleEngine.model.Agent;
 
-import signals.DayChoiceSignal;
+import com.mindsmiths.armory.ArmoryAPI;
+import com.mindsmiths.armory.components.DescriptionComponent;
+import com.mindsmiths.armory.components.ImageComponent;
+import com.mindsmiths.armory.components.PrimarySubmitButtonComponent;
+import com.mindsmiths.armory.components.TextAreaComponent;
+import com.mindsmiths.armory.components.TitleComponent;
+import com.mindsmiths.armory.templates.BaseTemplate;
+import com.mindsmiths.armory.templates.TemplateGenerator;
+import com.mindsmiths.armory.components.CloudSelectComponent;
 
+import models.OnboardingStage;
+import signals.DayChoiceSignal;
 
 @Data
-@ToString
+@ToString(callSuper = true)
 @NoArgsConstructor
-public class Ava extends Agent { 
+public class Ava extends Agent {
+    OnboardingStage onboardingStage;
 
     public Ava(String connectionName, String connectionId) {
         super(connectionName, connectionId);
     }
 
+    public void showScreen(BaseTemplate screen) {
+        ArmoryAPI.showScreen(getConnection("armory"), screen);
+    }
+
+    public void showScreens(String firstScreenId, Map<String, BaseTemplate> screens) {
+        ArmoryAPI.showScreens(getConnection("armory"), firstScreenId, screens);
+    }
+
+    public void showFamiliarityQuizScreens(){
+        Map<String, BaseTemplate> screens = new HashMap<String, BaseTemplate>();
+        Option[] buttons = Mitems.getOptions("onboarding.familiarity-quiz.buttons");
+        
+        //Adding intro screen
+        String introScreenTitle = Mitems.getText("onboarding.familiarity-quiz.introscreentitle");
```

Comment:
In mitems, all names should be human readable. So for example - "Intro screen title"